dask/utils.py

```diff
@@ -1124,12 +1124,23 @@ def is_arraylike(x):
     False
     """
     from .base import is_dask_collection
+    from .array.utils import IS_NEP18_ACTIVE
+
+    # Fallback to True for cases where NEP-18 isn't active (e.g. older versions of NumPy)
+    implements_array_fuction = (
+        hasattr(x, "__array_function__") if IS_NEP18_ACTIVE else True
+    )
 
     return bool(
         hasattr(x, "shape")
         and isinstance(x.shape, tuple)
         and hasattr(x, "dtype")
         and not any(is_dask_collection(n) for n in x.shape)
+        # We special case scipy.sparse arrays as having partial support for them is
+        # useful in scenerios where we mostly call `map_partitions` or `map_blocks`
+        # with scikit-learn functions on dask arrays and dask dataframes.
+        # https://github.com/dask/dask/pull/3738
+        and (implements_array_fuction or "scipy.sparse" in typename(type(x)))
```

Comment:
```suggestion
        and (implements_array_fuction or "scipy.sparse" in x_typename or "cupyx.scipy.sparse" in x_typename)
```

Reply:
My thinking was that `"scipy.sparse"` would cover both the actual `scipy.sparse` case as well as the CuPy sparse case. Is having two checks against `x_typename` just for code clarity?

Reply:
Ah good point. Wasn't thinking about it. Yeah that's fine, but maybe we could include that in the comment above?

Reply:
👍 just updated the comment to explicitly mentioned `cupyx.scipy.sparse` arrays

Reply:
Great thanks! Looks good 😄